framework/src/parser/Parser.C

```diff
@@ -637,10 +637,14 @@ Parser::parse(const std::vector<std::string> & input_filenames)
 
   for (auto & input_filename : _input_filenames)
   {
-    MooseUtils::checkFileReadable(input_filename, true);
-
-    std::ifstream f(input_filename);
-    std::string input((std::istreambuf_iterator<char>(f)), std::istreambuf_iterator<char>());
+    // Parse the input text string if non-empty, otherwise read file from disk
+    std::string input(input_text);
+    if (input.empty())
+    {
+      MooseUtils::checkFileReadable(input_filename, true);
+      std::ifstream f(input_filename);
+      input = std::string((std::istreambuf_iterator<char>(f)), std::istreambuf_iterator<char>());
+    }
```

Comment:
This logic is weird to me. I think that if you provide multiple filenames... and an input text, it would read the input text multiple times... yeah?

Reply:
Yes, this is true and admittedly a bit weird.

The idea here is that the `_input_text` parameter is private and hidden in the MooseApp.

And this option is only used by the MooseServer to remove disk IO from input processing.

The MooseServer also only sets a single `input_filename` when it sets the `_input_text`.

So there would never be more than one `input_filename` when `input_text` is also used.

But this not a good assumption for future-proofing using the private `_input_text` option.

I can add this check to the top of the `parse()` method to prevent that case you describe.

```cpp
if (!input_text.empty() && input_filenames.size() != 1)
  mooseError("If 'input_text' is provided, then 'input_filenames' must contain a single filename");
```

There will currently be no way to hit that error in practice, but it will guard it for the future.

Is this an okay check to stop the unintended case of reading `input_text` multiple times?

Reply:
This isn't pretty, but I don't think it's a significant concern